pkg/controller/datavolume/controller-base.go

```diff
@@ -219,8 +220,12 @@ func addDataVolumeControllerCommonWatches(mgr manager.Manager, dataVolumeControl
 
 func getDataVolumeOp(dv *cdiv1.DataVolume) dataVolumeOp {
 	src := dv.Spec.Source
+
 	if (src != nil && src.PVC != nil) || dv.Spec.SourceRef != nil {
-		return dataVolumeClone
+		return dataVolumePvcClone
+	}
+	if (src != nil && src.Snapshot != nil) || dv.Spec.SourceRef != nil {
+		return dataVolumeSnapshotClone
```

Comment:
The `|| dv.Spec.SourceRef != nil` in both ifs makes it order dependent. Are you sure that's what we want here?

Reply:
nice catch, nope!
we must GET datasource in here now to distinct between pvc/snap?

anyway sourceref support will be in a follow up so I'll add a fixme for now

Reply:
added fixme